code/nuxeo-apidoc-jsf-webengine/src/main/java/org/nuxeo/apidoc/seam/browse/SeamApiBrowser.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * (C) Copyright 2006-2010 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Thierry Delprat
+ */
+package org.nuxeo.apidoc.seam.browse;
+
+import java.util.List;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+
+import org.nuxeo.apidoc.browse.Distribution;
+import org.nuxeo.apidoc.plugin.PluginSnapshot;
+import org.nuxeo.apidoc.seam.api.SeamComponentInfo;
+import org.nuxeo.apidoc.seam.plugin.SeamPlugin;
+import org.nuxeo.apidoc.snapshot.DistributionSnapshot;
+import org.nuxeo.apidoc.snapshot.SnapshotManager;
+import org.nuxeo.ecm.webengine.model.Resource;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.ecm.webengine.model.impl.DefaultObject;
+import org.nuxeo.runtime.api.Framework;
+
+@WebObject(type = "seam")
+public class SeamApiBrowser extends DefaultObject {
+
+    protected String distributionId;
+
+    protected boolean embeddedMode = false;
+
+    @Override
+    protected void initialize(Object... args) {
+        distributionId = (String) args[0];
+        if (args.length > 1) {
+            Boolean embed = (Boolean) args[1];
+            embeddedMode = embed != null && embed;
+        }
+    }
+
+    protected SnapshotManager getSnapshotManager() {
+        return Framework.getService(SnapshotManager.class);
+    }
+
+    @GET
+    @Produces("text/html")
+    @Path("listSeamComponents")
```

Comment:
For a proper REST API; I'd recommend: `/components` here to GET components or even better at the `/` lvl if there is no other resources than `component` in this object.

Reply:
This is the legacy URL for the listing of components. Now there's an extra "seam" in the URL compared to before but i'm not planning on refactoring all these, they just helped me design the plugin extension point on the platform side.

That being said if you see a way to remove the extra "seam" i'll take it but i don't think it's possible with the webengine fragment behaviour.

Sample old URLs:
https://explorer.nuxeo.com/nuxeo/site/distribution/Nuxeo%20Platform%20LTS%202019-10.10/listSeamComponents
https://explorer.nuxeo.com/nuxeo/site/distribution/Nuxeo%20Platform%20LTS%202019-10.10/viewSeamComponent/seam:actionContextProvider

New URLs will look like:
https://explorer.nuxeo.com/nuxeo/site/distribution/Nuxeo%20Platform%20LTS%202019-10.10/seam/listSeamComponents
https://explorer.nuxeo.com/nuxeo/site/distribution/Nuxeo%20Platform%20LTS%202019-10.10/seam/viewSeamComponent/seam:actionContextProvider